esphome/components/online_image/online_image.cpp

```diff
@@ -126,9 +132,17 @@ void OnlineImage::update() {
   }
   accept_header.value = accept_mime_type + ",*/*;q=0.8";
 
+  if (!this->etag_.empty()) {
+    headers.push_back(http_request::Header{IF_NONE_MATCH_HEADER_NAME, this->etag_});
+  }
+
+  if (!this->last_modified_.empty()) {
+    headers.push_back(http_request::Header{IF_MODIFIED_SINCE_HEADER_NAME, this->last_modified_});
+  }
+
   headers.push_back(accept_header);
 
-  this->downloader_ = this->parent_->get(this->url_, headers);
+  this->downloader_ = this->parent_->get(this->url_, headers, {ETAG_HEADER_NAME, LAST_MODIFIED_HEADER_NAME});
```

Comment:
IMHO the code would be slightly better readable, if the collect headers were extracted to a named constant:
```suggestion
  const auto collect_headers{ETAG_HEADER_NAME, LAST_MODIFIED_HEADER_NAME};
  this->downloader_ = this->parent_->get(this->url_, headers, collect_headers);
```

Reply:
:+1: